controller/api/destination/test_util.go

```diff
@@ -0,0 +1,447 @@
+package destination
+
+import (
+	"testing"
+
+	pb "github.com/linkerd/linkerd2-proxy-api/go/destination"
+	"github.com/linkerd/linkerd2/controller/api/destination/watcher"
+	"github.com/linkerd/linkerd2/controller/api/util"
+	"github.com/linkerd/linkerd2/controller/k8s"
+	pkgk8s "github.com/linkerd/linkerd2/controller/k8s"
+	"github.com/sirupsen/logrus"
+	logging "github.com/sirupsen/logrus"
+)
+
+func makeServer(t *testing.T) *server {
```

Comment:
it looks like the functions here only use the T to abort when there's an error.  If we make these functions fallible (with an error in the return type) instead of taking a T as an argument, then it we can use them from the fuzzers without needing to construct a dummy T to pass in.

Reply:
We also use `t` to create the log used by all the watchers that are created (like [here](https://github.com/linkerd/linkerd2/blob/f4ffd50e026f075d1e727aade97423bad5889f42/controller/api/destination/test_util.go#L328)). I think having named log fields for these tests is good to have even if it means needing to pass in a `T`.